tldw_Server_API/tests/Chat/unit/test_chat_service_tool_autoexec.py

```
@@ -432,6 +971,278 @@ async def save_message_fn(_db, _conv_id, payload, use_transaction=True):
+        "validate_structured_response",
+        lambda **_kwargs: (_ for _ in ()).throw(StructuredGenerationParseError("parse failed")),
+    )
+
+    metrics = _RunFirstMetrics()
+    saved_payloads: list[dict[str, Any]] = []
+
+    async def save_message_fn(_db, _conv_id, payload, use_transaction=True):
+        saved_payloads.append(payload)
+        return f"m-{len(saved_payloads)}"
+
+    with pytest.raises(Exception):
```

Ruff (0.15.7)

[warning] Do not assert blind exception: `Exception`

(B017)